web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx

```diff
@@ -300,7 +307,7 @@ function AppVersionHistoryRow(props: Props) {
               ) : preflightState.preflightState !== "" ? (
                 <>
                   <Link
-                    to={`/app/${app?.slug}/downstreams/${app?.downstream.cluster?.slug}/version-history/preflight/${version?.sequence}`}
+                    to={`/app/${app?.slug}/downstreams/${app?.downstream?.cluster?.slug}/version-history/preflight/${version?.sequence}`}
```

Comment:
could this be using selectedApp and downstream directly? 
```suggestion
                    to={`/app/${selectedApp?.slug}/downstreams/${downstream?.cluster?.slug}/version-history/preflight/${version?.sequence}`}
```